shell/server/core.js

```diff
@@ -198,7 +198,11 @@ class PersistentUiViewImpl {
         throw new Error("grain no longer exists");
       }
 
-      const pkg = Packages.findOne({ _id: grain.packageId });
+      let pkg = Packages.findOne({ _id: grain.packageId });
+      if (!pkg) {
+        pkg = DevPackages.findOne({ appId: grain.appId });
```

Comment:
I agree with David.

Also note that in theory if a DevPackage is present it should take priority over installed packages since starting the grain will actually open the DevPackage, not the installed package... however, I'm not sure I'd actually act on that theory since it would mean needlessly searching the DevPackages collection even on production servers, so I guess ignore this.